source/reference/array.rst

```diff
@@ -236,7 +236,136 @@ Unevaluated Items
+``unevaluatedItems`` like this:
+
+.. schema_example::
+
+    {
+      "$id": "https://example.com/my-extended-tuple",
+
+      "$ref": "https://example.com/my-tuple",
+      "prefixItems": [
+        true,
+        true,
+        { "type": "boolean" }
+      ],
+      "unevaluatedItems": false
+    }
+
+With this, you can use ``$ref`` to reference the first two
+``prefixItems`` and keep the schema "closed" to two arguments when
+you need it, "open" to more arguments when you need it. A reference to
+``/my-tuple#closed`` would disallow more than two items. And because
+``unevaluatedItems`` only sees inside its own subschema, if you
+want to add an item, add it inside that subschema.
+
+This means you can also put ``unevaluatedItems`` in a nested tuple.
+
+.. schema_example::
+
+    {
+      "prefixItems": [
+        { "type": "string" }
+      ],
+      "allOf": [
+        {
+          "prefixItems": [
+            true,
+            { "type": "number" }
+          ]
+        }
+      ],
+      "unevaluatedItems": false
+    }
+    --
+    ["foo", 42]
+    // All the array items are evaluated. The schema passes validation.
+    --X
+    ["foo", 42, null]
+    // The third value is unevaluated, so ``unevaluatedItems`` is true,
+    // not false. The schema fails validation.
+
+Note these two restraints, though. First, ``items`` doesn't "see
+inside" any instances of ``allOf``, ``anyOf``, or ``oneOf`` in the
+same subschema. In this next example, ``items`` ignores ``allOf`` and
+thus fails to validate.
+
+.. schema_example::
+
+    {
+      "allOf": [{ "prefixItems": [{ "enum": [1, "a"] }]  }],
+      "items": { "const": 2 }
+    }
+    --X
+    [1, "a", 2]
+
+Second, ``unevaluatedItems`` can't "see inside" cousins (vertically
+adjacent properties inside a separate pair of {curly braces} with the
+same "parent"— ``anyOf``, ``if``, ``not``, or similar). For instance,
+in the example below, the ``unevaluatedItems`` doesn't "see inside"
+the ``prefixItems`` cousin before it. All instances fail validation
+because ``"prefixItems": [ true ]`` matches only length 1 arrays, and
+``{ "unevaluatedItems": false }`` matches only empty arrays.
+
+.. schema_example::
+
+    {
+      "allOf": [
+        { "prefixItems": [true] },
+        { "unevaluatedItems": false }
+      ]
+    }
+    --X
+    [1]
+
+.. note::
+   For a tall list of more examples, read our `unevaluatedItems Test Suite <https://github.com/json-schema-org/JSON-Schema-Test-Suite/blob/main/tests/draft2020-12/unevaluatedItems.json>`_ on GitHub. We test a lot of use cases
+   there, including uncommon ones.
+
+Here are some of our examples in the suite:
+   * ``unevaluatedItems`` nested inside another ``unevaluatedItems``
+   * ``if/then/else`` statements interacting with ``unevaluatedItems``
+   * nested ``if/then/else`` statements interacting with ``unevaluatedItems``
+   * ``unevaluatedItems`` ignoring non-arrays
+   * ``unevaluatedItems`` interacting with the ``not`` keyword
+   * and more
```

Comment:
I'm not sure about the idea of linking to the test suite. My main concern is the links going out of date as the specification and the tests suite evolves. Another concern is consistency, we don't do that anywhere else in UJS. The other concern is that the audience of the test suite is very different that UJS. The audience of UJS is schema authors learning JSON Schema. The audience of the test suite is implementers. UJS shows how and when you might want to use a feature. The test suite is trying to break implementations.

The cross-linking could be useful, but I think I'm leaning toward preferring that it's not included.